mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -713,6 +713,15 @@ private[spark] object RandomForest extends Logging {
         binAggregates.metadata.numSplits(featureIndex) != 0
       }
 
+    // If no valid splits for features, then this split is invalid,
+    // return invalid information gain stats.  Take any split and continue.
+    if (validFeatureSplits.isEmpty) {
+      val parentImpurityCalculator = binAggregates.getImpurityCalculator(0, 0)
```

Comment:
Here, you're using feature index zero, even though this node might have a feature subset that does not include zero. The code still works, but it might be better to use `val dummyIndex = featuresForNode.map(_.head).getOrElse(0)`